packages/fxa-react/components/DataBlock/index.tsx

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, {
+  useCallback,
+  useState,
+  useEffect,
+  MouseEvent,
+  FocusEvent,
+} from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { copy } from 'fxa-react/lib/clipboard';
+
+export type DataBlockProps = {
+  value: string | string[];
+  copyable?: boolean;
+  id?: string;
+  onCopied?: (copiedValue: string) => void;
+  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
+  onBlur?: (event: FocusEvent<HTMLButtonElement>) => void;
+  onFocus?: (event: FocusEvent<HTMLButtonElement>) => void;
+  readerText?: string;
+};
+
+export const DataBlock = ({
+  value,
+  id,
+  copyable,
+  onCopied,
+  onClick,
+  onBlur,
+  onFocus,
+  readerText,
+}: DataBlockProps) => {
+  const [focussed, setFocussed] = useState<boolean>(false);
+  const valueIsArray = Array.isArray(value);
+  const [inputId, setInputId] = useState<string | undefined>(id);
+  const readerId = `${inputId}-sr`;
+
+  async function copyToClipboard() {
+    const copyValue = valueIsArray
+      ? (value as string[]).join(', ')
+      : (value as string);
+
+    await copy(copyValue);
+    onCopied && onCopied(copyValue);
```

Comment:
Is this an optional callback for use later? I'm not sure why we need `dataBlockFocus` and `dataBlockBlur`

Reply:
> Is this an optional callback for use later?

Yes. We will likely be adding a tooltip or some other functionality so I want to expose this for future use.